### PR TITLE
fix: use encrypt api for all encryption and decryption

### PR DESCRIPTION
**Pull request by ArjunKarthik**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
encrypt/decrypt should happen only via encryption service and DEK should be same in application and encryption service

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
encrypt/decrypt should happen only via encryption service and DEK should be same in application and encryption service

## How did you test it?
Changes can be verified only through logs. Look for "Fall back to Application Encryption" or "Fall back to Application Decryption" in the logs post deployment after running below tests. If logs are present encryption/decryption failed with encryption service and fall back to the application encryption which has to be investigated.

Test with merchant created before the changes deployed
1. Create Payment
2. Retrieve Payment
3. Create Refund
4. Retrieve Payment
Test with merchant created after the changes deployed
1. Create Merchant
2. Create API Key
3. Create MCA
4. Create Payment
6. Retrieve Payment
7. Create Refund
8. Retrieve Payment
Requires sanity on all the flows

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by dracarys18 — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

File: crates/router/src/types/domain/user.rs [hunk 1]
```diff
@@ -988,18 +1001,6 @@ impl UserFromStorage {
...
-                        identifier: Identifier::User(key_store.user_id.clone()),
-                    },
-                )
-                .await
-                .change_context(UserErrors::InternalServerError)?;
-            }
-
             state
                 .global_store
                 .insert_user_key_store(key_manager_state, key_store, &master_key.to_vec().into())
```
Comment (ThisIsMani) [inline crates/router/src/types/domain/user.rs:1006]: What will happen to transferred key if the db insertion fails here? Will it also be deleted?

Comment (ArjunKarthik) [reply crates/router/src/types/domain/user.rs:1006]: No that won't be removed.

Comment (ThisIsMani) [reply crates/router/src/types/domain/user.rs:1006]: Will that doesn't cause inconsistency?

Comment (ArjunKarthik) [reply crates/router/src/types/domain/user.rs:1006]: No, if new user store created for the same user it will be updated in the key store

Comment (ThisIsMani) [reply crates/router/src/types/domain/user.rs:1006]: Discussed with @ArjunKarthik, came to a conclusion that this function is temporary and transfer will override the service.

**Review by ThisIsMani — APPROVED** [review]

**Pull request merged**